respy/solve.py

```diff
@@ -46,35 +46,31 @@ def solve(params, options, state_space):
     optim_paras, options = process_params_and_options(params, options)
 
     states = state_space.states
-    wages = state_space.get_attribute("wages")
-    nonpecs = state_space.get_attribute("nonpecs")
     is_inadmissible = state_space.get_attribute("is_inadmissible")
 
-    wages, nonpecs = _create_choice_rewards(
-        states, wages, nonpecs, is_inadmissible, optim_paras
-    )
+    wages, nonpecs = _create_choice_rewards(states, is_inadmissible, optim_paras)
     state_space.set_attribute("wages", wages)
     state_space.set_attribute("nonpecs", nonpecs)
 
-    if optim_paras["delta"] == 0:
-        expected_value_functions = _solve_for_myopic_individuals(
-            state_space.get_attribute("expected_value_functions")
-        )
-        state_space.set_attribute("expected_value_functions", expected_value_functions)
-    else:
-        state_space = _solve_with_backward_induction(state_space, optim_paras, options)
+    state_space = _solve_with_backward_induction(state_space, optim_paras, options)
 
     return state_space
 
 
 @parallelize_across_dense_dimensions
-def _create_choice_rewards(states, wages, nonpecs, is_inadmissible, optim_paras):
+def _create_choice_rewards(states, is_inadmissible, optim_paras):
     """Create wage and non-pecuniary reward for each state and choice.
 
     Note that missing wages filled with ones and missing non-pecuniary rewards with
     zeros. This is done in :meth:`_initialize_attributes`.
 
     """
+    n_states = states.shape[0]
+    n_choices = len(optim_paras["choices"])
+
```

Comment:
We create container only in here because there is no gain in doing so before?

Reply:
Before, I initialized some arrays including wages and nonpecs in the state space class and let this function overwrite the arrays. But, this is might be an anti-pattern for parallelization with dask :) as input arrays are normally treated read-only. I think we should revisit this again once the state space is reworked. Then, https://github.com/zarr-developers/zarr-python might be a promising candidate for memory mapped arrays.